lib/matplotlib/cbook/__init__.py

```diff
@@ -1469,6 +1475,7 @@ def violin_stats(X, method, points=100):
         - median: The median value for this column of data.
         - min: The minimum value for this column of data.
         - max: The maximum value for this column of data.
+        - quantiles: The quantile values for this column of data.
```

Comment:
I am confused. If user directly use this function as API, my code only extends the results that user can get. If this extension is considered as a "break", then I don't think it is necessary to make qunatiles key optional.

Reply:
If user never directly use this function, then it is only called by violinplot(). Should this be counted as a API though?

Reply:
Matplotlib is extensively used out in the wild. By experience, for every remote but possible way of using the official API, there's someone out there, who actually has code with it. Therefore, we've become quite sensitive to possible API breaks. They will pop up as bug reports sooner or later. So we have to at least document the changes.

Tuple unpacking of the result will break with the change:
~~~
coords, vals, mean, median, min, max = violin_stats()
~~~

Reply:
I agree to have a document. But I feel it is odd to make quantiles optional since the current code in violin_stats doesn't care whether those value will be drawn out.

Reply:
I'm sorry, I misread the documentation here. It's a list of dictionaries with one dictionary per input dataset. Adding keys to these dicts is not breaking the API. So the above API change note is not necessary and should be removed again. Sorry for the noise.